src/draw/lv_draw_image.c

```diff
@@ -116,6 +121,131 @@ lv_image_src_t lv_image_src_get_type(const void * src)
+        int32_t h = lv_area_get_height(coords);
+
+        _lv_image_buf_get_transformed_area(&draw_area, w, h, draw_dsc->rotation, draw_dsc->scale_x, draw_dsc->scale_y,
+                                           &draw_dsc->pivot);
+
+        draw_area.x1 += coords->x1;
+        draw_area.y1 += coords->y1;
+        draw_area.x2 += coords->x1;
+        draw_area.y2 += coords->y1;
+    }
+
+    lv_area_t clipped_img_area;
+    if(!_lv_area_intersect(&clipped_img_area, &draw_area, draw_unit->clip_area)) {
+        return;
+    }
+
+    lv_image_decoder_dsc_t decoder_dsc;
+    lv_result_t res = lv_image_decoder_open(&decoder_dsc, draw_dsc->src, NULL);
+    if(res != LV_RESULT_OK) {
+        LV_LOG_ERROR("Failed to open image");
+        return;
+    }
+
+    img_decode_and_draw(draw_unit, draw_dsc, &decoder_dsc, coords, &clipped_img_area, draw_core_cb);
+
+    lv_image_decoder_close(&decoder_dsc);
+}
+
+void _lv_draw_image_tiled_helper(lv_draw_unit_t * draw_unit, const lv_draw_image_dsc_t * draw_dsc,
+                                 const lv_area_t * coords, lv_draw_image_core_cb draw_core_cb)
+{
+    if(draw_core_cb == NULL) {
+        LV_LOG_WARN("draw_core_cb is NULL");
+        return;
+    }
+
+    lv_image_decoder_dsc_t decoder_dsc;
+    lv_result_t res = lv_image_decoder_open(&decoder_dsc, draw_dsc->src, NULL);
+    if(res != LV_RESULT_OK) {
+        LV_LOG_ERROR("Failed to open image");
+        return;
+    }
+
+    int32_t img_w = lv_area_get_width(coords);
+    int32_t img_h = lv_area_get_height(coords);
+
+    lv_area_t tile_area = *coords;
+    int32_t tile_x_start = tile_area.x1;
+
+    while(tile_area.y1 <= draw_unit->clip_area->y2) {
+        while(tile_area.x1 <= draw_unit->clip_area->x2) {
+
+            lv_area_t clipped_img_area;
+            if(_lv_area_intersect(&clipped_img_area, &tile_area, draw_unit->clip_area)) {
+                img_decode_and_draw(draw_unit, draw_dsc, &decoder_dsc, &tile_area, &clipped_img_area, draw_core_cb);
+            }
+
+            tile_area.x1 += img_w;
+            tile_area.x2 += img_w;
+        }
+
+        tile_area.y1 += img_h;
+        tile_area.y2 += img_h;
+        tile_area.x1 = tile_x_start;
+        tile_area.x2 = tile_x_start + img_w - 1;
+    }
+
+    lv_image_decoder_close(&decoder_dsc);
+}
+
 /**********************
  *   STATIC FUNCTIONS
  **********************/
+
+static void img_decode_and_draw(lv_draw_unit_t * draw_unit, const lv_draw_image_dsc_t * draw_dsc,
+                                lv_image_decoder_dsc_t * decoder_dsc,
+                                const lv_area_t * img_area, const lv_area_t * clipped_img_area,
+                                lv_draw_image_core_cb draw_core_cb)
+{
+    lv_draw_image_sup_t sup;
+    sup.alpha_color = draw_dsc->recolor;
+    sup.palette = decoder_dsc->palette;
+    sup.palette_size = decoder_dsc->palette_size;
+
+    /*The whole image is available, just draw it*/
+    if(decoder_dsc->decoded || decoder_dsc->img_data) {
```

Comment:
I'll update PR https://github.com/lvgl/lvgl/pull/4948 in these days and check if any better method can be used to detect available image areas.